sopel/modules/currency.py

```diff
@@ -79,10 +79,17 @@ def build_reply(amount, base, target, out_string):
     rate = float(rate_raw)
     result = float(rate * amount)
 
-    if target == 'BTC':
-        return out_string + ' {:.5f} {},'.format(result, target)
+    digits = 0
+    while 1 / 10**digits > result:
+        digits += 1
+        if digits >= 8:
+            break
```

Comment:
Since it's a while loop, the exit condition can be directly inside the while expression:

```suggestion
    # up to 10 (8+2) digits precision when result is less than 1
    # as smaller results need more precision
    while digits < 8 and 1 / 10**digits > result:
        digits += 1
```

And a bit of inline comment to help future readers to understand what is going on here.